features-sdk/step_definitions/controllers-steps.js

```diff
@@ -165,3 +166,41 @@ Then('I got an error with id {string}', function (id) {
 
   assert(this.props.error.id === id, `Expected error to have id "${id}", but got "${this.props.error.id}"`);
 });
+
+When('I call the deprecated publicApi method', async function () {
+  if (process.env.KUZZLE_PROTOCOL !== 'http') {
+    return;
+  }
+
+  if (process.env.NODE_ENV !== 'development') {
+    return;
+  }
+
+  const response = await this.sdk.query({
+    controller: 'server',
+    action: 'publicApi'
+  });
+
+  this.props.deprecations = response.deprecations;
+});
+
+Then('I should receive a deprecation notice', async function () {
```

Comment:
This step should be converted to a more generic step that allow to match the `response`

You could modify [this step](https://github.com/kuzzleio/kuzzle/blob/master/features-sdk/step_definitions/controllers-steps.js#L56) to allows match an array of object either in the `this.props.response` or in the `this.props.result` object.

The regex could be something like `/I should receive a ("(.*?)" )?array (of objects )?in the "(.*?)" matching:/`

```
Then I should receive a "deprecations" array of objects in the "response" matching:
```